### PR TITLE
Fix a few new `clippy` warnings

### PR DESCRIPTION
**Pull request by kaivol**
removed some unnecessary casts (`clippy::unnecessary-cast`) and ignored another warning (`clippy::manual_clamp`)

### CODE REVIEW

**Comment by ChrisDenton** [comment]
Is there a reason not to use clamp here?

**Comment by riverar** [comment]
Sounds reasonable. Can we also change the title of the PR to reflect what was changed?

**Comment by kennykerr** [comment]
> Is there a reason not to use clamp here?

The trouble is that it will panic if min > max.

File: crates/samples/direct2d/src/main.rs [hunk 1]
```diff
@@ -179,6 +179,7 @@ impl Window {
 
         let size = unsafe { target.GetSize() };
 
+        #[allow(clippy::manual_clamp)]
```
Comment (kennykerr) [inline crates/samples/direct2d/src/main.rs:182]: Seems like a `clippy` bug that this warning is not appropriate and must be suppressed.

**Pull request merged**